sched/adaptdl_sched/policy/pollux.py

```diff
@@ -40,6 +40,32 @@ def __init__(self):
         self._min_util = 0.35
         self._max_util = 0.65
 
+    # a simple allocate policy: allocate the new job 
+    # to the first available node. 
+    def _allocate_job(self, job_info, node_info_list):
+        job_key = list(job_info.keys())[0]
+        job_resources = job_info[job_key].resources
+        min_replicas = max(job_info[job_key].min_replicas,1)
+        node_list = []
+        for node_name, node in node_info_list.items():
+            # number of replica fit in this node
+            replica_this = min(node.resources.get(key,0) // val 
+                   for key, val in job_resources.items())
+            # we are done if addding this node gives enough replicas    
+            if (len(node_list) + replica_this) >= min_replicas:
+                replica_need = min_replicas - len(node_list) 
+                node_list.extend([node_name] * replica_need)
+                LOG.info("find %s for job %s with minimum replicas: %s",
+                         node_list, job_key, min_replicas)    
+                return node_list
+            # else we replicate as many as we can on this node
+            else:
```

Comment:
Hmm, this can violate the Pollux restriction that no two distributed jobs (i.e. each having replicas on multiple nodes) may share the same node (having a replica on the same node). Maybe it's best to just search for resources on a single node instead of allowing the new job to span multiple nodes.